pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -858,6 +858,38 @@ static void validateUpsertAndDedupConfig(TableConfig tableConfig, Schema schema)
       }
     }
 
+    // enableConsistentDeletes shouldn't exist with metadataTTL
+    if (upsertConfig != null && upsertConfig.isEnableConsistentDeletes()) {
```

Comment:
why not combine all the checks inside one if-check? as the if (condition) is same